python/paddle/fluid/tests/unittests/test_protobuf_descs.py

```diff
@@ -201,33 +201,13 @@ def test_remove_op(self):
         op1.set_type("test")
         op2.set_type("test")
 
-        var0 = block.var("var0")
-        var1 = block.var("var1")
-        var2 = block.var("var2")
-        var3 = block.var("var3")
-        var4 = block.var("var4")
-        var5 = block.var("var5")
-
-        op0.set_input("X", ["var0"])
-        op0.set_output("Y", ["var0"])
-        op1.set_input("X", ["var1", "var2"])
-        op1.set_output("Y", ["var3", "var4"])
-        op2.set_input("X", ["var1"])
-        op2.set_output("Y", ["var4", "var5"])
-
-        program.sync_with_cpp()
-
-        # remove op1, its input var2 and output var3 will be removed at the same time,
-        # but its input var1 and output var4 will not be removed since they are used for op2.
         block.remove_op(1, 2)
         program.sync_with_cpp()
```

Comment:
Does `sync_with_cpp` need to be updated to a simpler version that clear all ops and add back the descs?

Reply:
Discussed with @jacquesqiao, this simpler version is so dangerous since it will clear all ops. 

If we create an op1 first, then call `insert_op`, and then using op1 will fail. The reason is that all the instances in Python end are removed, they are changed to C++ instances instead.

Reply:
That make sense, I see `sync_with_cpp` have already deleted those `clear` operations.